### ISSUE TITLE
[web] MediaQuery highContrast is always false

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#104272]
`MediaQuery.highContrast` is not implemented on the web. We should implement it.

One way to do that is using a CSS media query, such as `html.window.matchMedia('force-colors: active')`. Since `matchMedia` is quite expensive (it parses an executes CSS code it's given), the value should be cached, but also the cache should be invalidated when the value changes.
